src/dotnet-aspnet-codegenerator/Program.cs

```diff
@@ -6,127 +6,218 @@
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
-using System.Reflection;
-using System.Runtime.Loader;
-using System.Threading;
-using Microsoft.CodeAnalysis;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli.Utils;
 using Microsoft.DotNet.ProjectModel;
-using Microsoft.DotNet.ProjectModel.Workspaces;
-using Microsoft.Extensions.CodeGeneration.DotNet;
-using Microsoft.Extensions.CodeGeneration.EntityFrameworkCore;
-using Microsoft.Extensions.CodeGeneration.Templating;
-using Microsoft.Extensions.CodeGeneration.Templating.Compilation;
 using Microsoft.Extensions.CommandLineUtils;
-using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
 
 namespace Microsoft.Extensions.CodeGeneration
 {
     public class Program
     {
         private static ConsoleLogger _logger;
+
         private const string APPNAME = "Code Generation";
         private const string APP_DESC = "Code generation for Asp.net Core";
 
+        private static bool isProjectDependencyCommand;
+
         public static void Main(string[] args)
         {
+
             Stopwatch stopWatch = new Stopwatch();
             stopWatch.Start();
-
             _logger = new ConsoleLogger();
+            // We don't want to expose this flag, as it only needs to be used for identifying self invocation. 
+            isProjectDependencyCommand = args.Contains("--no-dispatch");
```

Comment:
If you transition to using the `DotnetToolDispatcher` (will be getting it in today) you wont have to do this work.